cmd/pulumi-test-language/interface_test.go

```diff
@@ -85,6 +86,22 @@ func TestPolicyPacks(t *testing.T) {
 	}
 }
 
+// Ensure provider tests do use language providers and no other test does.
+func TestLanguageProviders(t *testing.T) {
+	t.Parallel()
+
+	for name, test := range tests.LanguageTests {
+		isProvider := strings.HasPrefix(name, "provider-")
+		if isProvider {
+			assert.NotEmpty(t, test.LanguageProviders,
+				"test name %s must use language providers", name)
+		} else {
+			assert.Empty(t, test.LanguageProviders,
+				"test name %s must not use language providers", name)
```

Comment:
Nit: This reads a lot clearer to me.
```suggestion
				"test %q must use language providers", name)
		} else {
			assert.Empty(t, test.LanguageProviders,
				"test %q must not use language providers", name)
```